### PR TITLE
Fixed IPAddressInUse on allocating deallocated ip

### PR DESCRIPTION
**Pull request by quadewarren**
JIRA:NCP-1898

### CODE REVIEW

File: quark/ipam.py [hunk 1]
```diff
@@ -565,10 +565,14 @@ def _allocate_ips_from_subnets(self, context, new_addresses, net_id,
                      new_addresses=new_addresses, ip_address=ip_address)))
 
         subnets = subnets or []
+        allocated_ips = [ip.get("address_readable") for ip in new_addresses]
         for subnet in subnets:
             if not subnet:
                 continue
 
+            if str(ip_address) in allocated_ips:
+                continue
```
Comment (xroot88) [inline quark/ipam.py]: Here, if the ip_address was found in the new_addresses list, we return new_addresses unmodified.
Is this intended?

Comment (quadewarren) [inline quark/ipam.py]: This behavior is unchanged from the previous code. new_addresses actually doesn't appear to serve a purpose in this function. I've been seeing a lot of code that appears unused, so it looks like this code was gutted.

**Comment by roaet** [comment]
LGTM +1

🐣

**Comment by xroot88** [comment]
+1

**Pull request merged**